pkgs/development/python-modules/troposphere/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, pythonOlder
+
+  # python dependencies
+, awacs
+, cfn-flip
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "troposphere";
+  version = "4.1.0";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "cloudtools";
+    repo = pname;
+    rev = version;
+    hash = "sha256-cAn4Hty5f/RsCnUA59CxtGrhRgzVyaHe5PuQOM6lwEQ=";
+  };
+
+  propagatedBuildInputs = [
+    cfn-flip
+  ] ++
+  lib.lists.optional (pythonOlder "3.8")
+    typing-extensions;
+
+  checkInputs = [
+    pytestCheckHook
+    awacs
+  ];
+
+  passthru.optional-dependencies = {
+    policy = [ awacs ];
+  };
+
+  # https://github.com/cloudtools/troposphere/blob/4.1.0/Makefile#L87-L88
+  checkPhase = "python setup.py test";
```

Comment:
```suggestion
```
No need for this if you use `pytestCheckHook`.

Reply:
Not exactly the same, `pytestCheckHook` runs `${python.interpreter} -m pytest`, which for some reason fails for one test, while `${python.interpreter} setup.py test` works. I removed `pytestCheckHook` and let the `${python.interpreter} setup.py test`

Reply:
Does `${python.interpreter} -m unittest discover` work? That's preferred over using `setup.py`.

Reply:
Yes, that works too, I just updated it